tss-esapi/src/abstraction/transient/key_attestation.rs

```diff
@@ -0,0 +1,208 @@
+    /// Public part of the key used to protect the credential
+    pub attesting_key_pub: PublicKey,
+}
+
+impl TransientKeyContext {
+    /// Get the data required to perform a MakeCredential
+    ///
+    /// # Parameters
+    ///
+    /// * `object` - the object whose TPM name will be included in
+    /// the credential
+    /// * `key` - the key to be used to encrypt the secret that wraps
+    /// the credential
+    ///
+    /// **Note**: If no `key` is given, the default Endorsement Key
+    /// will be used.  
+    pub fn get_make_cred_params(
+        &mut self,
+        object: ObjectWrapper,
+        key: Option<ObjectWrapper>,
+    ) -> Result<MakeCredParams> {
+        let object_handle = self.load_key(object.params, object.material, None)?;
+        let (object_public, object_name, _) =
+            self.context.read_public(object_handle).or_else(|e| {
+                self.context.flush_context(object_handle.into())?;
+                Err(e)
+            })?;
+        self.context.flush_context(object_handle.into())?;
+
+        // Name of objects is derived from their publicArea, i.e. the marshaled TPMT_PUBLIC
+        let public = TPM2B_PUBLIC::from(object_public);
+        let mut pub_buf = [0u8; std::mem::size_of::<TPMT_PUBLIC>()];
+        let mut offset = 0;
+        let result = unsafe {
+            Tss2_MU_TPMT_PUBLIC_Marshal(
+                &public.publicArea,
+                &mut pub_buf as *mut u8,
+                std::mem::size_of::<TPMT_PUBLIC>()
+                    .try_into()
+                    .map_err(|_| Error::local_error(WrapperErrorKind::InternalError))?,
+                &mut offset,
+            )
+        };
+        let result = Error::from_tss_rc(result);
+        if !result.is_success() {
+            error!("Error in marshalling TPM2B");
+            return Err(result);
+        }
+        // `offset` will be small, so no risk in the conversion below
+        let public = pub_buf[..offset as usize].to_vec();
+
+        let attesting_key_pub = match key {
+            None => get_ek_object_public(&mut self.context)?,
+            Some(key) => key.material.public,
+        };
+        Ok(MakeCredParams {
+            name: object_name.value().to_vec(),
+            public,
+            attesting_key_pub,
+        })
+    }
+
+    /// Perform an ActivateCredential operation for the given object
+    ///
+    /// # Parameters
+    ///
+    /// * `object` - the object whose TPM name is included in the credential
+    /// * `key` - the key used to encrypt the secret that wraps the credential
+    /// * `credential_blob` - encrypted credential that will be returned by the
+    /// TPM
+    /// * `secret` - encrypted secret that was used to encrypt the credential
+    ///
+    /// **Note**: if no `key` is given, the default Endorsement Key
+    /// will be used. You can find more information about the default Endorsement
+    /// Key in the [ek] module.
+    pub fn activate_credential(
```

Comment:
It would be nice if this also had a code example like the `Context` functions. Copying something working from end to end tests is OK for first version IMHO. (doctests are a good way for newcomers to learn how to use the API)

Reply:
I'll add this as a TODO in #279 and will add examples for most/all of the methods in `TransientKeyContext`.

Reply:
Done